### PR TITLE
kds chapter 3 exercises

### PR DESCRIPTION
**Pull request by glaKris**

### CODE REVIEW

File: chapter03/kds/ex1.pl [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+use warnings;
+use strict;
+
+my @lines;
+
+print "Type a few lines of text then exit with ctrl + d:\n";
+do{
+  @lines = <STDIN>;
+}while( <> );
+
```
Comment (JPeace2) [inline chapter03/kds/ex1.pl:12]: I'm not sure why you need the do while loop.

File: chapter03/kds/ex2.pl [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+chomp( @numbers );
+
+foreach $num ( @numbers ){
+  if( !looks_like_number( $num ) ){
+    print "$num is not a number, ignoring...\n";
+  } elsif( !( $num <= 6 && $num >= 0 ) ){
+    print "$num is not between 0-6, ignoring...\n";
+  } else {
+    print "$names[$num]\n";
+  }
```
Comment (JPeace2) [inline chapter03/kds/ex2.pl:25]: I numbering was supposed to go between 1 and 7.

File: chapter03/kds/ex3.pl [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+my @strings;
+
+print "Type a few strings on separate lines, exit with ctrl + d:\n";
+do{
+  @strings = <STDIN>;
+}while( <> );
+
+print "Type y/Y to display on one line:\n";
+if( <STDIN> =~ /[yY]/ ){
```
Comment (pratyp) [inline chapter03/kds/ex3.pl:14]: Like the use of choice here.

File: chapter03/kds/ex2.pl [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+}while( <> );
+
+chomp( @numbers );
+
+foreach $num ( @numbers ){
+  if( !looks_like_number( $num ) ){
+    print "$num is not a number, ignoring...\n";
+  } elsif( !( $num <= 6 && $num >= 0 ) ){
+    print "$num is not between 0-6, ignoring...\n";
+  } else {
```
Comment (pratyp) [inline chapter03/kds/ex2.pl:23]: Nice decent validation. :+1:

**Comment by pratyp** [comment]
Looks good :+1:

File: chapter03/kds/ex2.pl (hunk 2 above)
Comment (qianliok) [inline chapter03/kds/ex2.pl:25]: yeah, number should start from 1

File: chapter03/kds/ex3.pl (hunk 3 above)
Comment (ardavey) [inline chapter03/kds/ex3.pl:14]: Amusingly, this would also match "no way!" :wink:

**Pull request merged**